### PR TITLE
Add test case for test_recurring_jobs_when_volume_detached_unexpectedly

### PR DESCRIPTION
**Pull request by khushboo-rancher**
https://github.com/longhorn/longhorn/issues/1922

Signed-off-by: khushboo-rancher <khushboo@rancher.com>

### CODE REVIEW

File: manager/integration/tests/test_recurring_job.py [hunk 1]
```diff
@@ -636,4 +683,64 @@ def test_recurring_jobs_when_volume_detached_unexpectedly():  # NOQA
...
+    wait_for_volume_detached(client, vol_name)
+
+    deployment['spec']['replicas'] = 1
+    apps_api.patch_namespaced_deployment(body=deployment,
+                                         namespace='default',
+                                         name=deployment["metadata"]["name"])
+    wait_deployment_replica_ready(apps_api, deployment["metadata"]["name"], 1)
+    pod_names = common.get_deployment_pod_names(core_api, deployment)
+
+    assert read_volume_data(core_api, pod_names[0], 'default') is not None
```
Comment (PhanLe1010) [inline manager/integration/tests/test_recurring_job.py:742]: Should we compare the checksum of the data inside volume to make sure that they are not corrupted?

File: manager/integration/tests/test_recurring_job.py [hunk 2]
```diff
@@ -636,4 +683,64 @@ def test_recurring_jobs_when_volume_detached_unexpectedly():  # NOQA
...
+            "task": "backup",
+            "retain": 1
+        }
+    ]
+    vol.recurringUpdate(jobs=jobs)
+
+    wait_for_recurring_backup_to_start(client, core_api, vol_name, 1)
+
+    crash_engine_process_with_sigkill(client, core_api, vol_name)
+    time.sleep(30)
```
Comment (PhanLe1010) [inline manager/integration/tests/test_recurring_job.py:726]: I do not quite understand why we need to sleep here?

File: manager/integration/tests/test_recurring_job.py [hunk 3]
```diff
@@ -636,4 +683,64 @@ def test_recurring_jobs_when_volume_detached_unexpectedly():  # NOQA
...
+    wait_for_volume_attached(client, vol_name)
```
Comment (PhanLe1010) [inline manager/integration/tests/test_recurring_job.py:727]: Is it better to wait for the volume to be in the attached state or to be healthy as in the test skeleton?

File: manager/integration/tests/test_recurring_job.py [hunk 4]
```diff
@@ -55,6 +64,45 @@ def wait_until_begin_of_an_even_minute():
...
+            snapshots = volume.snapshotList()
+        except (AttributeError, ApiException):
+            time.sleep(RETRY_BACKUP_INTERVAL)
+            continue
+        if len(snapshots) == n + 1:
+            for snapshot in snapshots:
+                if snapshot.children['volume-head']:
+                    snapshot_name = snapshot.name
+            break
+    assert snapshots is not None
```
Comment (PhanLe1010) [inline manager/integration/tests/test_recurring_job.py:86]: Why do we need to find the snapshot in this function?
I thought that we just need to check `v.backupStatus` and wait until it is. >= progress %
There is only 1 backup cronjob running so it is not possible that the backup process causes by a different job.

**Review by PhanLe1010 — COMMENTED** [review]
Remember to change the test skeleton in the end if we change the actual test

File: manager/integration/tests/test_recurring_job.py (hunk 3 above)
Comment (meldafrawi) [reply manager/integration/tests/test_recurring_job.py:727]: fixed

File: manager/integration/tests/test_recurring_job.py (hunk 4 above)
Comment (meldafrawi) [reply manager/integration/tests/test_recurring_job.py:86]: `wait_for_backup_completion` will require the snapshot name as a parameter, so it's efficient to get it while we're checking backup operation is started.

File: manager/integration/tests/test_recurring_job.py (hunk 2 above)
Comment (meldafrawi) [reply manager/integration/tests/test_recurring_job.py:726]: fixed

File: manager/integration/tests/test_recurring_job.py (hunk 1 above)
Comment (meldafrawi) [reply manager/integration/tests/test_recurring_job.py:742]: fixed

**Comment by meldafrawi** [comment]
@PhanLe1010, I've fixed all your comments and added them to a different PR https://github.com/longhorn/longhorn-tests/pull/530

@khushboo-rancher Great job, Thank you.

**Comment by meldafrawi** [comment]
Closing this PR in favor of https://github.com/longhorn/longhorn-tests/pull/530

**Pull request closed without merging**